docs/source/extending/WritingTIAndContextProviders.rst

```diff
@@ -0,0 +1,460 @@
+        "ipv4-geo": _OTXParams(path="/api/v1/indicators/IPv4/{observable}/geo"),
+        "ipv6-geo": _OTXParams(path="/api/v1/indicators/IPv6/{observable}/geo"),
+        ...
+
+This allows users to request the specific dataset for the IoC using
+the ``query_type`` parameter:
+
+.. code:: python3
+
+    tilookup.lookup_ioc("123.4.56.78", query_type="passivedns")
+
+
+Having decided on the keys needed, you can create the APILookupParams
+instances to tell the TILookup module how to form the HTTP requests.
+
+The :py:class:`APILookupParams <msticpy.context.http_provider.APILookupParams>` class
+has the following attributes:
+
+.. code:: python3
+
+    class APILookupParams:
+        """HTTP Lookup Params definition."""
+
+        path: str = ""
+        verb: str = "GET"
+        full_url: bool = False
+        headers: Dict[str, str] = Factory(dict)
+        params: Dict[str, Union[str, int, float]] = Factory(dict)
+        data: Dict[str, str] = Factory(dict)
+        auth_type: str = ""
+        auth_str: List[str] = Factory(list)
+        sub_type: str = ""
+
+The value of each item in the queries dictionary should be an
+instance of an ``APILookupParams`` class or one derived from it.
+
+.. note:: APILookupParams is an attrs class, if you create a subclass
+    from it you should also make that an attrs class.
+
+Several of the values of this class can have substitutable parameters
+where runtime values (e.g. the observable value)
+are inserted before making the request.
+
+**path**
+
+The sub-path for the query for this IoC type. This will be
+appended to the ``_BASE_URL`` value.
+
+.. code:: python3
+
+    _QUERIES = _QUERIES = {
+        "ipv4": APILookupParams(path="/api/v1/indicators/IPv4/{observable}/general"),
+
+In this example you can see that "{observable}", the IoC value, is
+a substitutable parameter.
+
+**verb**
+
+The provider framework currently only supports "GET"
+
+**full_url**
+If True, the ``_BASE_URL`` value is ignored and the ``path`` value
+is treated as a full URL and used in the request as-is.
+
+**headers**
+
+A dictionary of request headers.
+This also supports parameter substitution of any value surrounded with
+braces.
+
+Example:
+
+.. code:: python3
+
+     _QUERIES = _QUERIES = {
+        "ipv4": APILookupParams(
+            path="/api/v1/indicators/IPv4/{observable}/general"
+            headers = {"X-OTX-API-KEY": "{AuthKey}"}
+        ),
+        ...
+
+**params**
+A dictionary of request parameters. This also supports
+parameter substitution of values:
+
+.. code:: python3
+
+     _QUERIES = _QUERIES = {
+        "ipv4": APILookupParams(
+            path="/api/v1/indicators/IPv4"
+            params={"iocValue": "{observable}"},
+        ),
+        ...
+
+**data**
+This is currently not supported but we will implement if and
+when required. This is a dictionary that will be supplied as request data. supports
+parameter substitution for values.
+
+**auth_type**
+Currently only "HTTPBasic" is supported.
```

Comment:
This isn't limited to HTTP basic auth in the technical sense (ie, b64 encoded username:password). It can handle any URL parameter or HTTP header based authentication if I am reading the docs correctly.

Reply:
Ok. If you can suggest better wording, that would be great. Just that the auth param we supply to httpx is "HTTPBasic" vs. a bunch of other mechanism like client cert. Agree that the text is prob misleading.